cylc/flow/task_job_mgr.py

```diff
@@ -979,8 +979,8 @@ def _set_retry_timers(
             rtconfig = itask.tdef.rtconfig
 
         submit_delays = (
-            rtconfig['submission retry delays']
-            or itask.platform['submission retry delays']
+            rtconfig.get('submission retry delays', [])
```

Comment:
@MetRonnie  this one isn't, but I think it should be. Try running the workflow in the test without it. Needs to be ported to master.

Reply:
But the CI can! https://github.com/cylc/cylc-flow/actions/runs/7556705094/job/20574317445?pr=5927

Reply:
This fix does not appear to be on master / #5712

Reply:
You are right that it is not on master - I thought it had gone in with #5712, but it hasn't. 

However, I'm claiming this is the fix to bug1